File: 2021/08/25/main.go
package main

func judgeSquareSum(c int) bool {
	for i := 2; i*i <= c; i++ {
		if c%i != 0 {
			continue
		}
		cnt := 0
		for c%i == 0 {
			cnt++
			c /= i
		}
		if i%4 == 3 && cnt%2 != 0 {
			return false
		}
	}
	return c%4 != 3
}

File: 2021/08/25/main.py
class Solution:
    def judgeSquareSum(self, c: int) -> bool:
        i = 2
        while i * i <= c:
            cnt = 0
            if c % i == 0:
                while c % i == 0:
                    cnt += 1
                    c /= i
                if i % 4 == 3 and cnt % 2 != 0:
                    return False
            i += 1
        return c % 4 != 3
